File: sort/quick_sort.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
#include <stdio.h>
using namespace std;

/* Function to print an array */
void printArray(int arr[], int size) {
  int i;
  for (i = 0; i < size; i++)
    printf("%d ", arr[i]);
  printf("\n");
}

// } Driver Code Ends
class Solution {
public:
  // Function to sort an array using quick sort algorithm.
  void quickSort(int arr[], int low, int high) {
    // code here
    if (low >= high)
      return;
    int mid = partition(arr, low, high);
    quickSort(arr, low, mid - 1);
    quickSort(arr, mid + 1, high);
  }

public:
  int partition(int arr[], int low, int high) {
    // Your code here
    int pivot = arr[low];
    int i = low + 1;
    while (i <= high) {
      if (arr[i] <= pivot)
        ++i;
      else if (arr[i] > pivot) {
        swap(arr[i], arr[high--]);
      }
    }
    swap(arr[low], arr[high]);
    return high;
  }
};

//{ Driver Code Starts.
int main() {
  int arr[1000], n, T, i;
  scanf("%d", &T);
  while (T--) {
    scanf("%d", &n);
    for (i = 0; i < n; i++)
      scanf("%d", &arr[i]);
    Solution ob;
    ob.quickSort(arr, 0, n - 1);
    printArray(arr, n);
  }
  return 0;
}
// } Driver Code Ends

File: sort/quick_sort.py
# User function Template for python3


class Solution:
    # Function to sort a list using quick sort algorithm.
    def quickSort(self, arr, low, high):
        # code here
        if low >= high:
            return
        mid = self.partition(arr, low, high)
        self.quickSort(arr, low, mid - 1)
        self.quickSort(arr, mid + 1, high)

    def partition(self, arr, low, high):
        # code here
        j = low
        for i in range(low + 1, high + 1):
            if arr[i] <= arr[low]:
                j += 1
                arr[i], arr[j] = arr[j], arr[i]

        arr[low], arr[j] = arr[j], arr[low]
        return j


# {
# Driver Code Starts
# Initial Template for Python 3

if __name__ == "__main__":
    t = int(input())
    for i in range(t):
        n = int(input())
        arr = list(map(int, input().split()))
        Solution().quickSort(arr, 0, n - 1)
        for i in range(n):
            print(arr[i], end=" ")
        print()

# } Driver Code Ends
